skimage/filters/ridges.py

```diff
@@ -286,6 +298,13 @@ def sato(image, sigmas=range(1, 10, 2), black_ridges=True):
     if np.any(sigmas < 0.0):
         raise ValueError('Sigma values less than zero are not valid')
 
+    if mode is None:
+        warn("'constant' borders handling mode is deprecated and will be "
```

Comment:
First sentence here (and similar warning below) sounds a little off but that might just be the non-native speaker in me talking. Either use `'constant' mode is deprecated...` or `'constant' border handling is deprecated ...`? :wink:

Reply:
I agree that the wording is a bit awkward, but honestly, the constant mode is so terrible for edge filters, that I would call our previous behaviour a bug, and here just leave `mode='reflect'` as the default, then add a note in the release notes API changes that the border mode in certain filters has been updated to remove artifacts, and to use `mode='constant'` to recover the old behaviour.

What do others think here? @scikit-image/core

Reply:
At any rate, it's not 'constant' border mode that deprecated, since we didn't have a border mode before. The message should say something like "previously, this function implicitly used 'constant' as the border mode when dealing with the edge of the array. The new behavior is 'reflect'. To recover the old behavior, use mode='constant'". Something like that.

Reply:
I agree, I just added the warning here because `sato` and `hessian` were not suffering from the border artifacts like `frangi` and `meijering` did, so I did not considered that it was a bug. A simple warning like @jni is suggesting looks good to me.

Reply:
+!

Reply:
Just fwiw, my suggestion here was not to warn, but rather to put this note in the release notes. @stefanv, too much?

Reply:
It feels like that's a bit hidden; I presume few enough people read that, and those who do forget what it said after a few days.

Reply:
The workflow I was thinking was that essentially everyone would just be fine with this change, but people whose tests would fail would go and look at the release notes and find the culprit, which I've done a few times. But, I'm happy to leave it as a warning for now.

Reply:
>  the constant mode is so terrible for edge filters, that I would call our previous behaviour a bug, and here just leave mode='reflect' as the default, then add a note in the release notes API changes that the border mode in certain filters has been updated to remove artifacts, and to use mode='constant' to recover the old behaviour.

Sorry for not commenting earlier y'all, but I think this warning is too verbose... I'd go for a warning on the API change – since this is kinda of a big deal, shouldn't it be a note at the docstrings instead? I think this is something that would annoy more than inform, as it is.

Reply:
@alexdesiqueira I disagree! We have actually made a conscious effort to move our warnings and errors towards more verbosity. The idea is to contain all the information you need to fix the warning within the warning itself, rather than having to hunt around for a fix. I think that is the least annoying path, and I believe @stefanv agrees.